kalite/updates/api_views.py

```diff
@@ -129,10 +130,12 @@ def start_video_download(request):
     lang = json.loads(request.body or "{}").get("lang", "en")
 
     youtube_ids = get_download_youtube_ids(paths, language=lang)
-
+    ids_to_queue = {id_: title for (id_, title) in youtube_ids.iteritems()
```

Comment:
files_complete shows how many files have been downloaded. Each video
currently only has one associated file.

For a water tight check, ensuring that files_complete is less than
total_files would only give you undownloaded content.

But, do this in the method above to reduce database queries rather than
filtering after the fact.

On Fri, 18 Mar 2016, 15:11 Michael Gallaspy, <notifications@github.com>
wrote:

> In kalite/updates/api_views.py
> <https://github.com/learningequality/ka-lite/pull/5016#discussion_r56707880>
> :
>
> > @@ -129,10 +130,12 @@ def start_video_download(request):
> >      lang = json.loads(request.body or "{}").get("lang", "en")
> >
> >      youtube_ids = get_download_youtube_ids(paths, language=lang)
> > -
> > +    ids_to_queue = {id_: title for (id_, title) in youtube_ids.iteritems()
>
> How can I ensure that a file isn't downloaded *again* if it's already on
> disk? Is this information contained somehow in extra_fields? Or can I
> infer it from files_complete? What is the significance of that field?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/learningequality/ka-lite/pull/5016/files/64890447515a84b6c411b4f68d0ad1e383aae1b5#r56707880>
>